### PR TITLE
Update build system

### PR DESCRIPTION
**Pull request by impl**
This switches us to GitHub Actions, and retains the existing functionality minus the magic that nobody liked from `semantic-release`.

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+        key: ${{ runner.os }}-go-${{ hashFiles('**/go.sum') }}
+        restore-keys: |
+          ${{ runner.os }}-go-
+    - name: Install tools
+      env:
+        GO111MODULE: 'on'
+      run: |
+        cd
+        go get \
+          gotest.tools/gotestsum \
```
Comment (kyleterry) [inline .github/workflows/ci.yml:25]: we should consider adding these to `tools.go` in the root so they are tracked with `go.mod`.

**Review by hunner — APPROVED** [review]
LGTM

File: .github/workflows/ci.yml (hunk 1 above)
Comment (impl) [reply .github/workflows/ci.yml:25]: Ah yeah, I forgot about that pattern!

Comment (kyleterry) [reply .github/workflows/ci.yml:25]: :^)

**Pull request merged**